src/librustc_mir/shim.rs

```diff
@@ -213,10 +221,10 @@ fn build_drop_shim<'a, 'tcx>(tcx: ty::TyCtxt<'a, 'tcx, 'tcx>,
 }
 
 pub struct DropShimElaborator<'a, 'tcx: 'a> {
```

Comment:
If you want to make this "public", I would move it to a `util::drop_ladder` module, along with a `pub fn drop_ladder` to avoid the generator drop confusion.